// From directory with this definition, run this to regenerate python libraries:
// python -m grpc_tools.protoc --python_out=.. --grpc_python_out=.. -I . faucetconfrpc.proto

syntax = "proto3";

package faucetconfserver;

service FaucetConfServer {
  rpc GetConfigFile(GetConfigFileRequest) returns (GetConfigFileReply) {}
  rpc SetConfigFile(SetConfigFileRequest) returns (SetConfigFileReply) {}
  rpc DelConfigFromFile(DelConfigFromFileRequest) returns (DelConfigFromFileReply) {}
}

message GetConfigFileRequest {
  string config_filename = 1;
}

message GetConfigFileReply {
  string config_yaml = 1;
}

message SetConfigFileRequest {
  string config_filename = 1;
  string config_yaml = 2;
  bool merge = 3;
}

message SetConfigFileReply {
}

message DelConfigFromFileRequest {
  string config_filename = 1;
  string config_yaml_keys = 2;
}

message DelConfigFromFileReply {
}